app/src/main/java/androidx/core/app/SafeJobIntentService.kt

```diff
@@ -0,0 +1,28 @@
+package androidx.core.app
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import org.ligi.tracedroid.logging.Log
+import java.lang.SecurityException
+
+/**
+ * Mitigates the effect of a SecurityException which can occur in [JobIntentService].
+ * The issue is tracked here: https://issuetracker.google.com/issues/63622293
+ */
+abstract class SafeJobIntentService : JobIntentService() {
+
+    companion object{
+        private const val LOG_TAG = "SafeJobIntentService"
+    }
+
+    private val logging: Logging = Logging.get()
+
+    internal override fun dequeueWork(): GenericWorkItem? {
+        try {
+            return super.dequeueWork()
+        } catch (e: SecurityException) {
+            // There is not much we can do here.
+            logging.report(LOG_TAG, "$e")
+        }
+        return null
```

Comment:
Why not?:
``` kotlin
       return try {
            return super.dequeueWork()
        } catch (e: SecurityException) {
            // There is not much we can do here.
            logging.report(LOG_TAG, "$e")
            null
        }
```

Reply:
Cause that's basically my Code but less readable, at least IMO

Reply:
One of returns was too much...look again...this is actually a bit more idiomatic in terms of Kotlin and it follows the structural programming paradigms better.

Reply:
Ah, sorry, I overlooked that